### PR TITLE
Add missing text to buttons on the party / guild invite modal - Fixes #8391

### LINKED ISSUE #8391: Submit button in create/invite party modal has no label.
[//]: # (Before logging this issue, look through common problems at https://github.com/HabitRPG/habitrpg/issues If you find your issue there, read at least the first post to see if there is a workaround for you)

[//]: # (Github is primarily used for reporting bugs. If you have a feature request, use "Help > Request a Feature" so that the feature request can be vetted by the larger Habitica community)

[//]: # (To report a bug in one of the mobile apps, please report it in the correct repository. Android: https://github.com/HabitRPG/habitrpg-android, iOS: https://github.com/HabitRPG/habitrpg-ios)

[//]: # (For more guidelines see https://github.com/HabitRPG/habitrpg/issues/2760)

[//]: # (Fill out relevant information - UUID is found in Settings -> API)
General Info
  * UUID: a654bd1f-e925-438b-8d98-9ac4462b973f
  * Browser: Firefox 50
  * OS: Ubuntu Linux

### Description
[//]: # (Describe bug in detail here. Include pictures if helpful.)

The submit button in the create/invite party modal has no label. This causes it to look like a blue strip of unknown purpose. See this image for an example:

party_submit_issue

This also happens in Chromium 55 on Ubuntu.

### PR DESCRIPTION
**Pull request by eanplatter**
[//]: # (Note: See http://habitica.wikia.com/wiki/Using_Habitica_Git#Pull_Request for more info)

[//]: # (Put Issue # or URL here, if applicable. This will automatically close the issue if your PR is merged in)
Fixes #8391 

### Changes
[//]: # (Describe the changes that were made in detail here. Include pictures if necessary)
When #8328 was added, the text for the buttons was changed from =env.t('sendInvitations') to the dynamic message influenced by the groupsCtrl {{group.sendInviteText}} which makes sense to show the different button text.

The only problem is that if the button isn't subject to the fee text, then the group has no concept of sendInviteText.

All I did was went to where the group object originates groupService.js and added a default sendInviteText.

### Before
screen shot 2017-01-11 at 11 51 44 pm

### After
screen shot 2017-01-11 at 11 49 57 pm

[//]: # (Put User ID in here - found in Settings -> API)

----
UUID: 29664a36-68e0-4d97-a2b4-9343ebbfa8a6

### CODE REVIEW

**Comment by TheHollidayInn** [comment]
Looks good!

Although, I think the issue here is if the modal is launched via the `inviteOrStartParty ` method there is no default. There is still a default if `openInviteModal ` is called and the group is not subject to the fee.

So, wouldn't you mind doing two things for this PR:
1. Abstracting the common code between the openInviteModal and InviteOrStartParty to the groupService
2. Add a test in the groupService spec to ensure we don't regress.

**Comment by eanplatter** [comment]
@TheHollidayInn thanks for the insight, I'll take a look at this after work and push up an update!

**Comment by eanplatter** [comment]
Oh also, @TheHollidayInn is there any documentation on creating a user with a subscription, so I can test the add user with sub state?

**Comment by TheHollidayInn** [comment]
Sounds good! 

You won't need to create a subscription. You can just mock the data inside of the tests. Also, it is a group subscription not a user subscription. They are slightly different and the information is stored on the Group model.

File: test/client-old/spec/services/groupServicesSpec.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 'use strict';
 
 describe('groupServices', function() {
-  var $httpBackend, $http, groups, user;
+  var $httpBackend, $http, groups, user, $rootScope;
```
Comment (eanplatter) [inline test/client-old/spec/services/groupServicesSpec.js]: Added a `$rootScope` variable to be used in the inject method.

**Comment by eanplatter** [comment]
@TheHollidayInn do you mind looking at my work? This stack is almost 100% foreign to me, so idk if I am doing it right :P 

I attempted to combine the logic of opening the modal for invites into the groupsServices and removed the extra code from the groupsCtrl.

I then wrote a test to check if it is a guild, it adds text for the buttons, and same for party. 

Thanks!

File: test/client-old/spec/services/groupServicesSpec.js [hunk 2]
```diff
@@ -166,4 +168,32 @@ describe('groupServices', function() {
...
+      },
+      purchased: {
+        plan: {
+          customerId: '123',
+        },
+      },
+    };
+
+    groups.inviteOrStartParty(guild);
+    expect(guild.sendInviteText).to.eql('Send Invitations +$3.00/month/user');
```
Comment (TheHollidayInn) [inline test/client-old/spec/services/groupServicesSpec.js]: You should be able to use the translation library here.

File: website/client-old/js/controllers/groupsCtrl.js [hunk 3]
```diff
@@ -2,6 +2,7 @@
 
 habitrpg.controller("GroupsCtrl", ['$scope', '$rootScope', 'Shared', 'Groups', '$http', '$q', 'User', 'Members', '$state', 'Notification',
   function($scope, $rootScope, Shared, Groups, $http, $q, User, Members, $state, Notification) {
+    $scope.inviteOrStartParty = Groups.inviteOrStartParty;
```
Comment (TheHollidayInn) [inline website/client-old/js/controllers/groupsCtrl.js]: You changed the function name here, which will cause an issue. openInviteModal is different than inviteOrStartParty although they share much logic.

File: website/views/options/social/group.jade [hunk 4]
```diff
@@ -87,7 +87,7 @@ a.pull-right.gem-wallet(ng-if='group.type!="party"', popover-trigger='mouseenter
             h3.panel-title
               =env.t('members')
               span(ng-if='group.type=="party" && (group.onlineUsers || group.onlineUsers == 0)')= ' (' + env.t('onlineCount', {count: "{{group.onlineUsers}}"}) + ')'
-              button.pull-right.btn.btn-primary(ng-click="openInviteModal(group)")=env.t("inviteFriends")
+              button.pull-right.btn.btn-primary(ng-click="inviteOrStartParty(group)")=env.t("inviteFriends")
```
Comment (TheHollidayInn) [inline website/views/options/social/group.jade]: Ah, nevermind. I see you change the function name here.

File: website/client-old/js/services/groupServices.js [hunk 5]
```diff
@@ -220,19 +220,25 @@ angular.module('habitrpg')
 
     function inviteOrStartParty (group) {
       Analytics.track({'hitType':'event','eventCategory':'button','eventAction':'click','eventLabel':'Invite Friends'});
+      
+      var sendInviteText = window.env.t('sendInvitations');
+      if (group.type !== 'party' && group.type !== 'guild') {
```
Comment (TheHollidayInn) [inline website/client-old/js/services/groupServices.js:225]: Good job here.

**Comment by TheHollidayInn** [comment]
Looks great to me! Just one change in the tests.

File: test/client-old/spec/services/groupServicesSpec.js (hunk 2 above)
Comment (eanplatter) [reply test/client-old/spec/services/groupServicesSpec.js]: Ah good point, I'll patch this tonight, thanks!

**Comment by eanplatter** [comment]
Updated! Thanks for the feedback @TheHollidayInn

**Comment by eanplatter** [comment]
@TheHollidayInn Is this ready to merge? I don't want the PR to get stale XD

**Comment by TheHollidayInn** [comment]
Noted towards your Tier progress! Congrats on tier 1. Thank you :D

**Comment by eanplatter** [comment]
Wahoo, thanks man!

**Pull request merged**